tcms/testcases/migrations/0010_remove_bug.py

```diff
@@ -17,6 +17,10 @@ def forward_copy_data(apps, schema_editor):
         )
 
 
+def backward_empty_callable(apps, schema_editor):
+    pass
+
+
```

Comment:
When the test logs says `Rolling back testcases.0010_remove_bug`, just before it fails it means rolling back **TO** the specified migration, i.e. the specified migration is the last that will be applied. Everything after it (in the migrations order will not be applied) which is `0011_trim_bugsystem_fields.py`. 

0011 contains RemoveField operations which removes NOT NULL fields, e.g. `validate_reg_ex` (see the section in the log above the last one for the actual operations that get performed).

So at the end of the transaction all of the removed columns will be brought back and there are existing records in the DB. Now these new columns have null values b/c there's nothing to populate them and hence the failure. 

One way which may work is to add a RunPython migration in 0011 as the first one - empty forwards & a very simple backwards which will restore the required field values. Ideally the forwards migration will store the values somewhere under /tmp and the backwards operation will try to restore exactly the same values as were before.

Reply:
Not quite sure if this will work, will there be access to the tmp file during the backwards operation? Unless I don't quite understand your suggestion.

Reply:
I've tried what I think you meant but doesn't move things along.

Reply:
May not be reversible if `validate_reg_exp` was not nullable [going by this.](https://docs.djangoproject.com/en/3.0/ref/migration-operations/#removefield)